src/file_types.py

```diff
@@ -0,0 +1,339 @@
+"""
+Copyright 2023 NOAA
+All rights reserved.
+
+Collection of methods to facilitate handling of score db requests
+
+"""
+
+from collections import namedtuple
+import copy
+from dataclasses import dataclass, field
+from datetime import datetime
+import json
+import pprint
+from db_action_response import DbActionResponse
+import score_table_models as stm
+from score_table_models import FileType as ft
+import time_utils
+import db_utils
+
+from pandas import DataFrame
+import sqlalchemy as db
+from sqlalchemy.dialects.postgresql import insert
+from sqlalchemy.inspection import inspect
+from sqlalchemy import and_, or_, not_
+from sqlalchemy import asc, desc
+from sqlalchemy.sql import func
+
+FileTypeData = namedtuple(
+    'FileTypeData',
+    [
+        'name',
+        'file_template',
+        'file_format',
+        'stat_type',
+        'description'
+    ],
+)
+
+@dataclass
+class FileType:
+    ''' file type object storing data related to the file info '''
+    name: str
+    file_template: str
+    file_format: str
```

Comment:
Is there a reason for us to have a separate file_format column? (in addition to file_template?)

Reply:
I was envisioning file template as a reference to the name of the file (i.e. ocean-%Y%m.nc) but the format would be netcdf, text, CSV, or something along those lines. I can see that being relevant in future in regards to harvesters designed for certain file formats. Or if it's a customized file like the text logs are output in csv.

Reply:
OK. i just couldn't see a use case for a search under `file_format`. It's up to you.